vowpalwabbit/parse_regressor.cc

```diff
@@ -50,9 +50,51 @@ template <class T> class random_weights_wrapper
       *iter = (float)(merand48(index) - 0.5);
     }
 };
+// box-muller polar implementation
+template <class T> class polar_normal_weights_wrapper
+{
+public:
+    static void func(typename T::iterator& iter)
```

Comment:
This isn't the Box-Muller I'm familiar with?  This one: https://en.wikipedia.org/wiki/Box%E2%80%93Muller_transform#Basic_form doesn't need the Do{...} While